File: rss2json.py
from flask import Flask, jsonify
import requests
import xmltodict

app = Flask(__name__)

PODCAST_URL = 'https://anchor.fm/s/49f0c604/podcast/rss'

def fetch_podcast_data():
    try:
        # Fazendo a requisição para o feed RSS
        response = requests.get(PODCAST_URL)
        response.raise_for_status()
        return response.text
    except requests.exceptions.RequestException as e:
        print(f"Erro ao buscar o feed RSS: {e}")
        return None

def parse_podcast_data(xml_text):
    # Converte o XML em um dicionário usando xmltodict
    podcast_dict = xmltodict.parse(xml_text)
    items = podcast_dict['rss']['channel']['item']

    podcast_data = []
    for item in items:
        title = item['title']
        enclosure_url = item['enclosure']['@url']
        image_url = item.get('itunes:image', {}).get('@href', 'Imagem indisponível')

        podcast_data.append({
            'title': title,
            'enclosureUrl': enclosure_url,
            'imageUrl': image_url
        })

    return podcast_data

@app.route('/podcast', methods=['GET'])
def get_podcast():
    # Busca e analisa os dados do podcast
    xml_text = fetch_podcast_data()
    
    if not xml_text:
        return jsonify({'error': 'Erro ao buscar o feed do podcast'}), 500

    podcast_data = parse_podcast_data(xml_text)
    
    # Retorna os dados do podcast em formato JSON
    return jsonify(podcast_data)

if __name__ == '__main__':
    app.run(debug=True)

File: api/rss2json.mjs
// api/podcast.js
import { DOMParser } from "xmldom";

const url = "https://anchor.fm/s/49f0c604/podcast/rss";

async function fetchPodcastData(url) {
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error(`Erro na requisição: ${response.status}`);
  }
  return response.text();
}

function parsePodcastData(xmlText) {
  const parser = new DOMParser();
  const xmlDoc = parser.parseFromString(xmlText, "text/xml");
  const items = xmlDoc.getElementsByTagName("item");

  return Array.from(items).map((item) => {
    const name =
      item.getElementsByTagName("title")[0]?.textContent ||
      "Título indisponível";
    const url =
      item.getElementsByTagName("enclosure")[0]?.getAttribute("url") ||
      "URL indisponível";
    const cover_art_url =
      item.getElementsByTagName("itunes:image")[0]?.getAttribute("href") ||
      "Imagem indisponível";

    return { name, url, cover_art_url };
  });
}

async function getPodcastData() {
  try {
    const xmlText = await fetchPodcastData(url);
    const podcastData = parsePodcastData(xmlText);
    return podcastData;
  } catch (error) {
    console.error("Erro ao buscar dados do podcast:", error);
    return { error: error.message };
  }
}

export default async (req, res) => {
  const podcastData = await getPodcastData();
  res.status(200).json(podcastData);
};
